cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -157,6 +163,177 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
+        }
+
+        static void broadcast(wrapped_type & arr_out, pybind11::array & arr_in)
+        {
+            if (dtype_is_type<bool>(arr_in))
+            {
+                TypeBroadCastImpl<bool>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<int8_t>(arr_in))
+            {
+                TypeBroadCastImpl<int8_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<int16_t>(arr_in))
+            {
+                TypeBroadCastImpl<int16_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<int32_t>(arr_in))
+            {
+                TypeBroadCastImpl<int32_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<int64_t>(arr_in))
+            {
+                TypeBroadCastImpl<int64_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<uint32_t>(arr_in))
+            {
+                TypeBroadCastImpl<uint32_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<uint16_t>(arr_in))
+            {
+                TypeBroadCastImpl<uint16_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<uint32_t>(arr_in))
+            {
+                TypeBroadCastImpl<uint32_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<uint64_t>(arr_in))
+            {
+                TypeBroadCastImpl<uint64_t>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<float>(arr_in))
+            {
+                TypeBroadCastImpl<float>::broadcast(arr_out, arr_in);
+            }
+            else if (dtype_is_type<double>(arr_in))
+            {
+                TypeBroadCastImpl<double>::broadcast(arr_out, arr_in);
+            }
+            else
+            {
+                throw std::runtime_error("input array data type not support!");
+            }
+        }
+
+        static void throw_shape_error(wrapped_type & arr_out, pybind11::array & arr_in)
+        {
+            const auto & left_shape = arr_out.shape();
+            const auto * right_shape = arr_in.shape();
+
+            std::ostringstream msg;
+            msg << "Broadcast input array from shape(";
+            for (pybind11::ssize_t i = 0; i < arr_in.ndim(); ++i)
+            {
+                msg << right_shape[i];
+                if (i != arr_in.ndim() - 1)
+                {
+                    msg << ", ";
+                }
+            }
+            msg << ") into shape(";
+            for (size_t i = 0; i < arr_out.ndim(); ++i)
+            {
+                msg << left_shape[i];
+                if (i != arr_out.ndim() - 1)
+                {
+                    msg << ", ";
+                }
+            }
+            msg << ")";
+
+            throw std::runtime_error(msg.str());
+        };
+    };
+
+    template <typename D /* for destination type */>
+    struct TypeBroadCastImpl
```

Comment:
In the beginning, I want to just implement a template `broadcast_impl<D>` in the origin `TypeBroadCast` struct, but I found I couldn't do that (error: "use of function template name with no prior declaration in function call with explicit template")

Reply:
If you have code like that, post to gist and I can help analyze.